OpenSim/Sandbox/MatlabScripts/c3d_reading.m

```diff
@@ -1,124 +1,158 @@
+adapter = C3DFileAdapter();
 
-tables = adapter.read('test_walking.c3d')
+tables = adapter.read('test_walking.c3d');
 
-
-%% get the Markers
+%% get the marker data
 markers = tables.get('markers');
+% get the number of markers and frames
+nMarkers = markers.getNumColumns();
+nMarkerFrames = markers.getNumRows();
+% get the data rate of the markers. The Meta data is created from the c3d,
+% so the keys are not generic. In this case, we know we are looking for
+% 'DataRate'
+MarkerMetaDataKeys = markers.getTableMetaDataKeys();
+for i = 0 : MarkerMetaDataKeys.size() - 1
+
+    if strcmp( char(MarkerMetaDataKeys.get(i)), 'DataRate')
+        MarkerRate = str2num(markers.getTableMetaDataAsString(...   
+                                               MarkerMetaDataKeys.get(i)));
+        break
+    end
+end
 
-markerdata = opensimTimeSeriesTableToMatlab(markers);
-markerlabels = fieldnames(markerdata);
-
-% Print the (unrotated) markers to trc file
-trcfileadapter = TRCFileAdapter();
-trcfileadapter.write(markers,'test_walking.trc');
-
-%% Get the force
-% forces = tables.get('forces');
-% forcedata = opensimVec3TableToStruct(forces);
-% markerlabels = fieldnames(forcedata);
-% 
-% % Print the (unrotated) markers to trc file
-% trcfileadapter = TRCFileAdapter();
-% trcfileadapter.write(markers,'test_walking.trc');
-
+%% get the force data
+forces = tables.get('forces');
+% get the numner of forces and rows
+nForces = forces.getNumColumns();
+nForceFrames = forces.getNumRows();
+% get the data rate of the forces. The Meta data is created from the c3d,
+% so the keys are not generic. In this case, we know we are looking for
+% 'DataRate'
+ForceMetaDataKeys = forces.getTableMetaDataKeys();
+for i = 0 : ForceMetaDataKeys.size() - 1
+
+    if strcmp( char(ForceMetaDataKeys.get(i)), 'DataRate')
+        ForceRate = str2num(forces.getTableMetaDataAsString(...   
+                                               ForceMetaDataKeys.get(i)));
+        break
+    end
+end
 
 %% Define a rotation matix
-Rot = 90;          
-rotationMatrix = [1,0,0;0,cos(Rot*pi/180),-(sin(Rot*pi/180));0,sin(Rot*pi/180),cos(Rot*pi/180)];
-        
-%% Rotate marker data
-
-new_table = TimeSeriesTableVec3()
-
-
-TimeSeriesTableVec3::DependentsMetaData marker_dep_metadata{};
-marker_dep_metadata.setValueArrayForKey("labels", marker_labels);
-marker_table.setDependentsMetaData(marker_dep_metadata);
-
-double time_step{1.0 / acquisition->GetPointFrequency()};
-for f = 1:0 
-    
-    
-    rowVector = RowVector()
-    
-    
-    
-    for(auto it = marker_pts->Begin();
-        it != marker_pts->End();
-        ++it) {
-        auto pt = *it;
-        row[m++] = SimTK::Vec3{pt->GetValues().coeff(f, 0),
-                               pt->GetValues().coeff(f, 1),
-                               pt->GetValues().coeff(f, 2)};
-    }
-
-    
-marker_table.appendRow(0 + f * time_step, row);
+Rot = -90;
+rotationMatrix = [1,0,0;0,cosd(Rot),-(sin(Rot*pi/180));0,sin(Rot*pi/180),cosd(Rot)];
```

Comment:
> It would be great if OpenSim::Vec3tables had a method to rotate. That would make this script super fast.

Can you use a [SimTK::Transform()](https://simtk.org/api_docs/simbody/3.5/singletonSimTK_1_1Transform__.html)? Not sure whether all the necessary classes/methods are wrapped, though.